File: testdata/jennies/builders/builder_delegation/PHPBuilder/src/BuilderDelegation/DashboardLinkBuilder.php
<?php

namespace Grafana\Foundation\BuilderDelegation;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\BuilderDelegation\DashboardLink>
 */
class DashboardLinkBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\BuilderDelegation\DashboardLink $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\BuilderDelegation\DashboardLink();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\BuilderDelegation\DashboardLink
     */
    public function build()
    {
        return $this->internal;
    }

    public function title(string $title): static
    {
        $this->internal->title = $title;
    
        return $this;
    }

    public function url(string $url): static
    {
        $this->internal->url = $url;
    
        return $this;
    }

}

File: testdata/jennies/builders/builder_delegation/JavaBuilder/builder_delegation/DashboardLinkBuilder.java
package builder_delegation;


public class DashboardLinkBuilder implements cog.Builder<DashboardLink> {
    protected final DashboardLink internal;
    
    public DashboardLinkBuilder() {
        this.internal = new DashboardLink();
    }
    public DashboardLinkBuilder title(String title) {
        this.internal.title = title;
        return this;
    }
    
    public DashboardLinkBuilder url(String url) {
        this.internal.url = url;
        return this;
    }
    public DashboardLink build() {
        return this.internal;
    }
}
